Созданы тесты,добавлены фикстуры и генерация данных, сформирован Allure-report

dreamQA opened the pull request.

Commit:
Созданы тесты,добавлены фикстуры и генерация данных, сформирован Allure-report

DolgoarshinnykhLG commented on data.py:
```diff
@@ -0,0 +1,34 @@
+    login = faker.name()
+    email = faker.email()
+
+class Urls:
+    MAIN_URL = "https://stellarburgers.nomoreparties.site"
+    API_CREATE_USER = "/api/auth/register"
+    API_LOGIN = "/api/auth/login"
+    API_ORDER = "/api/v1/orders"
+    API_DELETE_USER = "/api/auth/user"
+    API_GET_USER = "/api/auth/user"
+    API_GET_INGREDIENTS = "/api/ingredients"
+    API_CREATE_ORDERS = "/api/orders"
```
Можно лучше: адреса лучше вынести из модуля тестов во внешний (например, urls). Они могут измениться, что затруднит поддержку тестов, плюс будет легче поддерживать несколько стендов если использовать конкатенацию адресов с базовым (MAIN_URL)

DolgoarshinnykhLG commented on tests/test_create_order.py:
```diff
@@ -0,0 +1,41 @@
+    def test_create_order_with_auth(self,create_user_and_get_token,get_ingredient_hash):
+        requests.post(f'{Urls.MAIN_URL}{Urls.API_LOGIN}',data= create_user_and_get_token)
+        ingredients = {'ingredients':[get_ingredient_hash['data'][1]['_id'],get_ingredient_hash['data'][4]['_id'],get_ingredient_hash['data'][8]['_id']]}
+        response = requests.post(f'{Urls.MAIN_URL}{Urls.API_CREATE_ORDERS}',data=ingredients)
```
Нужно исправить здесь и далее: необходимо сделать allure-аннотации для каждого метода отправки запроса. При такой реализации лучше всего подходит контекст-менеджер with allure.step

DolgoarshinnykhLG commented on tests/test_create_order.py:
```diff
@@ -0,0 +1,41 @@
+        ingredients = {'ingredients':[get_ingredient_hash['data'][1]['_id'],get_ingredient_hash['data'][4]['_id'],get_ingredient_hash['data'][8]['_id']]}
+        response = requests.post(f'{Urls.MAIN_URL}{Urls.API_CREATE_ORDERS}',data=ingredients)
+        order = response.json()
+        assert order['name'] == Burgers.SPICY_UNDEAD_CRATER_BURGER and order['success'] == True
```
Необходимо исправить здесь и далее: в тестах стоит также проверять и код ответа

DolgoarshinnykhLG commented on tests/test_create_user.py:
```diff
@@ -0,0 +1,42 @@
+    @allure.title('Проверка повторной регистрации пользователя')
+    def test_create_user_double(self):
+        payload = generate_user()
+        requests.post(f'{Urls.MAIN_URL}{Urls.API_CREATE_USER}', data=payload)
```
Можно лучше здесь и далее: шаг с созданием курьера дублируется в начале некоторых тестов и является их предусловием. Можно вынести его в фикстуру

DolgoarshinnykhLG commented on tests/test_create_user.py:
```diff
@@ -0,0 +1,42 @@
+        response = requests.post(f'{Urls.MAIN_URL}{Urls.API_CREATE_USER}', data=payload)
+        assert 403 == response.status_code and response.json()['message'] == ErrorMessage.TEXT_CREATE_403_DOUBLE
+
+    @allure.title('Проверка регистрации пользователя без почты')
+    def test_create_user_without_email(self):
+        payload = generate_user()
+        del payload['email']
+        response = requests.post(f'{Urls.MAIN_URL}{Urls.API_CREATE_USER}', data={})
+        assert 403 == response.status_code and response.json()['message'] == ErrorMessage.TEXT_CREATE_403_WRONG
+
+    @allure.title('Проверка регистрации пользователя без имени')
+    def test_create_user_without_name(self):
+        payload = generate_user()
+        del payload['name']
+        response = requests.post(f'{Urls.MAIN_URL}{Urls.API_CREATE_USER}', data={})
+        assert 403 == response.status_code and response.json()['message'] == ErrorMessage.TEXT_CREATE_403_WRONG
+
+    @allure.title('Проверка регистрации пользователя без пароля')
+    def test_create_user_without_password(self):
+        payload = generate_user()
+        del payload['password']
+        response = requests.post(f'{Urls.MAIN_URL}{Urls.API_CREATE_USER}', data={})
+        assert 403 == response.status_code and response.json()['message'] == ErrorMessage.TEXT_CREATE_403_WRONG
```
Можно лучше: Шаги тестов очень схожи. Можно их объединить с помощью параметризации

Commit:
Исправил только критические замечания,из-за нехватки времени с 6мес. ребенком
1.Нужно исправить здесь и далее: необходимо сделать allure-аннотации для каждого метода отправки запроса. При такой реализации лучше всего подходит контекст-менеджер with allure.step
2.Необходимо исправить здесь и далее: в тестах стоит также проверять и код ответа
